modules/macos_auto_puppet/files/auto-puppet.sh

```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Purpose: Bootstrap a macOS 14+ host from post-install (or image) to a complete Puppet run.
+
+export LANG=en_US.UTF-8
+export PATH="/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/puppetlabs/bin"
+
+# Puppet configuration
+LOCAL_PUPPET_REPO="/Users/relops/Desktop/puppet/ronin_puppet"
```

Comment:
The issue we ran into on linux is that when people change the GIT_REPO_URL in this script (or in the override), the git fetch or checkout below fail because it's not a fast-forward update or something. I fixed this for most cases on linux by using the GH username from the URL in the repo path (https://github.com/mozilla-platform-ops/ronin_puppet/blob/331a7b0fbfb8c3a1af94e9d14b564d8fa853cbcc/modules/puppet/templates/puppet-ubuntu-run-puppet.sh.erb#L37). I think there are other ways around it like telling pull/checkout to do non-fast-forwards or something.

Reply:
Nice. LGTM.